### ISSUE TITLE
Kani --visualize fails for tutorial example

### ISSUE THREAD

**Issue by celinval** [model-checking/kani#893]
I tried the following commands:
```
cd docs/src/tutorial/kinds-of-failure
# This will crash.
kani src/bounds_check.rs --function bound_check --visualize
# This will also crash.
cargo kani --function bound_check --visualize
```

I expected to see this happen: Have a generated report.

Instead, this happened:

```
--- begin invariant violation report ---
Invariant check failed
File: ../src/goto-instrument/cover_basic_blocks.cpp:189 function: update_source_lines
Condition: !str.empty()
Reason: source lines set must not be empty
Backtrace:
cbmc(print_backtrace(std::ostream&)+0x4e) [0x55e8230f2d1e]
cbmc(get_backtrace[abi:cxx11]()+0x16a) [0x55e8230f3caa]
cbmc(std::enable_if<std::is_base_of<invariant_failedt, invariant_failedt>::value, void>::type invariant_violated_structured<invariant_failedt, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x4f) [0x55e82295ecef]
cbmc(cover_basic_blockst::update_source_lines(cover_basic_blockst::block_infot&)+0x1d9) [0x55e8229e43e9]
cbmc(cover_basic_blockst::cover_basic_blockst(goto_programt const&)+0x240) [0x55e8229e65a0]
cbmc(+0x1e21d6) [0x55e8229d01d6]
cbmc(instrument_cover_goals(cover_configt const&, symbol_tablet const&, goto_functionst&, message_handlert&)+0x343) [0x55e8229d1013]
cbmc(cbmc_parse_optionst::process_goto_program(goto_modelt&, optionst const&, messaget&)+0x25d) [0x55e8229606fd]
cbmc(cbmc_parse_optionst::get_goto_program(goto_modelt&, optionst const&, cmdlinet const&, ui_message_handlert&)+0x519) [0x55e822961099]
cbmc(cbmc_parse_optionst::doit()+0x459) [0x55e822965689]
cbmc(parse_options_baset::main()+0xa8) [0x55e82295cc68]
cbmc(main+0x34) [0x55e82294b674]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fb37589fbf7]
cbmc(_start+0x2a) [0x55e82295e85a]


--- end invariant violation report ---
Traceback (most recent call last):
  File "/usr/local/bin/cbmc-viewer", line 11, in <module>
    sys.exit(viewer())
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/viewer.py", line 172, in viewer
    coverage = coveraget.make_coverage(args.viewer_coverage, args.srcdir, args.coverage)
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/coveraget.py", line 575, in make_coverage
    return CoverageFromCbmcXml(cbmc_coverage, srcdir)
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/coveraget.py", line 397, in __init__
    [load_cbmc_xml(xml_file, root) for xml_file in xml_files]
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/coveraget.py", line 397, in <listcomp>
    [load_cbmc_xml(xml_file, root) for xml_file in xml_files]
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/coveraget.py", line 403, in load_cbmc_xml
    xml = parse.parse_xml_file(xml_file, fail=True)
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/parse.py", line 17, in parse_xml_file
    return parse_xml_string(data.read(), xfile, fail)
  File "/usr/local/lib/python3.6/dist-packages/cbmc_viewer/parse.py", line 46, in parse_xml_string
    raise UserWarning(message) from None
UserWarning: Can't parse xml file src/bounds_check.coverage.xml: string <?xml version="1.0" encoding="UTF-8"?>
<...: no element found: line 200, column 0
Error: cbmc-viewer exited with status exit status: 1
```

**Comment by tautschnig** [comment]
Which CBMC version is this? The CBMC side of this should be fixed as of https://github.com/diffblue/cbmc/pull/6641, which is part of 5.51.0 and any later version. I tried with Kani rev 7152657415a and CBMC rev de9fa65099a78 and wasn't able to reproduce, so perhaps using a newer CBMC version will do the trick.

**Comment by celinval** [comment]
This has been fixed by #928 which upgrades CBMC to 5.52 (probably due to https://github.com/diffblue/cbmc/pull/6641).
